File: Frontend/Mobile/AwesomeProject/LoginScreen.js
import React, { useState } from 'react';
import { View, Text, TextInput, TouchableOpacity, Alert, Image, StyleSheet } from 'react-native';
import SignUpScreen from './SignUpScreen';

const LoginScreen = ({ userType }) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [showSignUp, setShowSignUp] = useState(false);

  const signIn = async () => {
    try {
      if (userType === 'Client') {
        // Perform client-specific login logic
        // ...

        // Navigate to ClientHomeScreen
        // navigation.navigate('ClientHome');
      } else if (userType === 'Employee') {
        // Perform employee-specific login logic
        // ...

        // Navigate to EmployeeHomeScreen
        // navigation.navigate('EmployeeHome');
      } else if (userType === 'Manager') {
        // Perform manager-specific login logic
        // ...

        // Navigate to ManagerHomeScreen
        // navigation.navigate('ManagerHome');
      }
    } catch (error) {
      Alert.alert('Sign In Error', 'Failed to sign in. Please check your credentials.');
    }
  };

  const navigateToSignUp = () => {
    setShowSignUp(true);
  };

  return (
    <View style={styles.container}>
      {showSignUp ? (
        <SignUpScreen />
      ) : (
        <>
          <Image source={require('./assets/medecin.png')} style={styles.logo} />
          <Text style={styles.title}>Login</Text>
          <View style={styles.inputContainer}>
            <TextInput
              style={styles.input}
              placeholder="Email (RPPS)"
              value={email}
              onChangeText={setEmail}
            />
          </View>
          <View style={styles.inputContainer}>
            <TextInput
              style={styles.input}
              placeholder="Password"
              secureTextEntry
              value={password}
              onChangeText={setPassword}
            />
          </View>
          <TouchableOpacity onPress={signIn} style={styles.signInButton}>
            <Text style={styles.buttonText}>Sign In</Text>
          </TouchableOpacity>
          <TouchableOpacity onPress={navigateToSignUp}>
            <Text style={styles.signUpText}>Sign up</Text>
          </TouchableOpacity>
        </>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#F9F9F9',
    width: '100%',
    height: '100%',
  },
  logo: {
    height: 200,
    width: 200,
    marginBottom: 20,
  },
  title: {
    fontSize: 40,
    fontWeight: 'bold',
    fontFamily: 'Roboto Condensed',
    marginBottom: 20,
  },
  inputContainer: {
    marginBottom: 10,
  },
  input: {
    backgroundColor: 'white',
    borderRadius: 12,
    paddingHorizontal: 20,
    paddingVertical: 10,
    width: 300,
    borderWidth: 1,
    borderColor: '#CBE4DE',
  },
  signInButton: {
    marginTop: 15,
    backgroundColor: '#0E8388',
    borderRadius: 12,
    paddingHorizontal: 20,
    paddingVertical: 16,
  },
  buttonText: {
    color: 'white',
    fontWeight: 'bold',
    fontSize: 18,
    textAlign: 'center',
  },
  signUpText: {
    marginTop: 20,
    fontSize: 16,
    textDecorationLine: 'underline',
  },
});

export default LoginScreen;

File: Frontend/Mobile/medzair_app/lib/LoginScreen.dart
import 'package:flutter/material.dart';
import 'package:medzair_app/medecin/medecin_homepage.dart';
import './SignUpScreen.dart';

class LoginScreen extends StatefulWidget {
  final String userType;

  LoginScreen({required this.userType});

  @override
  _LoginScreenState createState() => _LoginScreenState();
}

class _LoginScreenState extends State<LoginScreen> {
  TextEditingController emailController = TextEditingController();
  TextEditingController passwordController = TextEditingController();
  bool showSignUp = false;

  void signIn() {
    try {
      if (widget.userType == 'Client') {
        // Perform client-specific login logic
        // ...
        // Navigate to ClientHomeScreen
        // Navigator.pushNamed(context, 'ClientHome');
      } else if (widget.userType == 'Employee') {
        // Perform employee-specific login logic
        // ...
        // Navigate to EmployeeHomeScreen
        // Navigator.pushNamed(context, 'EmployeeHome');
      } else if (widget.userType == 'Manager') {
        // Perform manager-specific login logic
        // ...
        // Navigate to ManagerHomeScreen
        // Navigator.pushNamed(context, 'ManagerHome');
      }
    } catch (error) {
      showDialog(
        context: context,
        builder: (BuildContext context) {
          return AlertDialog(
            title: Text('Sign In Error'),
            content: Text('Failed to sign in. Please check your credentials.'),
            actions: <Widget>[
              TextButton(
                child: Text('OK'),
                onPressed: () {
                  Navigator.of(context).pop();
                },
              ),
            ],
          );
        },
      );
    }
  }

  void navigateToSignUp() {
    setState(() {
      showSignUp = true;
    });
  }

  void navigateToLogin() {
    setState(() {
      showSignUp = false;
    });
  }

  @override
  Widget build(BuildContext context) {
    if (showSignUp) {
      return SignUpScreen(
        onCancel: navigateToLogin,
      );
    }

    return Scaffold(
      body: Center(
        child: ListView(
          padding: EdgeInsets.symmetric(horizontal: 16.0, vertical: 120),
          children: <Widget>[
            Image.asset(
              'images/medecin.png',
              height: 200,
              width: 200,
              fit: BoxFit.contain,
              alignment: Alignment.center,
            ),
            SizedBox(height: 20),
            Text(
              'Login',
              style: TextStyle(
                fontSize: 40,
                fontWeight: FontWeight.bold,
                fontFamily: 'Roboto Condensed',
              ),
              textAlign: TextAlign.center,
            ),
            SizedBox(height: 20),
            TextField(
              decoration: InputDecoration(
                hintText: 'Email ou RPPS',
                border: OutlineInputBorder(),
              ),
              controller: emailController,
            ),
            SizedBox(height: 10),
            TextField(
              decoration: InputDecoration(
                hintText: 'Password',
                border: OutlineInputBorder(),
              ),
              obscureText: true,
              controller: passwordController,
            ),
            SizedBox(height: 20),
            ElevatedButton(
              onPressed: () {
                Navigator.push(
                  context,
                  MaterialPageRoute(
                    builder: (context) =>
                        const HomeMedecin(), // Replace with the appropriate route for the MissionsPage
                  ),
                );
              },
              child: Text('Sign In'),
            ),
            SizedBox(height: 20),
            TextButton(
              onPressed: navigateToSignUp,
              child: Text('Sign up'),
            ),
          ],
        ),
      ),
    );
  }
}
